### PR TITLE
Check that hdf5 has MPI if performing depletion with MPI

### PR DESCRIPTION
**Pull request by drewejohnson**
Check added in `openmc/depletion/__init__.py`. Without this check, the exporting of the `Results` to hdf5 will hang, as the second process attempts to write to a file that has already been opened on another process.

This error is only raised after a full transport calculation has been run. The check raises a more helpful error directly at the import from `openmc.deplete`, prior to transport calculations.

### CODE REVIEW

File: openmc/deplete/__init__.py [hunk 1]
```diff
@@ -6,10 +6,24 @@
...
 
 from .dummy_comm import DummyCommunicator
+
 try:
     from mpi4py import MPI
+
     comm = MPI.COMM_WORLD
     have_mpi = True
+    # check if running with MPI and if hdf5 is MPI-enabled
+    from h5py import get_config
```
Comment (paulromano) [inline openmc/deplete/__init__.py]: This import can/should be up top before any code logic. The only import we want to protect in the try/except block is for mpi4py. When you move it, also note that PEP8 specifies an [ordering for imports](https://www.python.org/dev/peps/pep-0008/#imports).

File: openmc/deplete/__init__.py [hunk 2]
```diff
@@ -6,10 +6,24 @@
...
+
+    if not get_config().mpi and comm.size > 1:
+        # Raise exception only on process 0
+        if comm.rank:
+            from sys import exit
```
Comment (paulromano) [inline openmc/deplete/__init__.py]: Same comment here, move the import to the top.

**Comment by drewejohnson** [comment]
Cleaned up the imports and some typos (hence the force pushes :/)

**Comment by paulromano** [comment]
Thanks @drewejohnson!

**Pull request merged**